### ISSUE TITLE
Unreliability

### ISSUE THREAD

**Issue by Bentechy66** [xMNG/Microsoft-Rewards-Bot#35]
I have a cronjob to run this script on my server with the arguments `--headless --mobile --pc --quiz --region UK`

And the script only goes few the first few of my accounts: 
```
2019-04-03 12:00:02,195 :: INFO :: root :: -----------------------New-----------                                                                                      -------------
2019-04-03 12:00:02,195 :: INFO :: root :: -------------------------------------                                                                                      -------------
2019-04-03 12:00:02,198 :: INFO :: root :: args parsed.
2019-04-03 12:00:02,201 :: INFO :: root :: logins retrieved.
2019-04-03 12:00:16,600 :: INFO :: root :: # of search items: 137

2019-04-03 12:00:16,601 :: INFO :: root :: -------------------------MOBILE------                                                                                      -------------------
2019-04-03 12:00:19,054 :: INFO :: root :: Logging in account2@gmail.com...
2019-04-03 12:00:43,341 :: INFO :: root :: Search Start
2019-04-03 12:01:15,685 :: ERROR :: root :: //*[@id="flyoutContent"] element Not                                                                                       Visible - Timeout Exception
2019-04-03 12:01:15,687 :: ERROR :: root :: //*[@id="flyoutContent"] cannot be l                                                                                      ocated.
Traceback (most recent call last):
  File "/srv/Microsoft-Rewards-Bot/ms_rewards.py", line 256, in wait_until_visib                                                                                      le
    WebDriverWait(browser, time_to_wait).until(ec.visibility_of_element_located(                                                                                      (by_, selector)))
  File "/usr/local/lib/python3.6/dist-packages/selenium/webdriver/support/wait.p                                                                                      y", line 80, in until
    raise TimeoutException(message, screen, stacktrace)
selenium.common.exceptions.TimeoutException: Message:

2019-04-03 12:01:16,262 :: INFO :: root :: WebDriverException while executing mo                                                                                      bile portion
Traceback (most recent call last):
  File "/srv/Microsoft-Rewards-Bot/ms_rewards.py", line 789, in <module>
    search(search_list, mobile_search=True)
  File "/srv/Microsoft-Rewards-Bot/ms_rewards.py", line 456, in search
    if get_point_total(mobile=True):
  File "/srv/Microsoft-Rewards-Bot/ms_rewards.py", line 662, in get_point_total
    pcsearch = browser.find_element_by_class_name('pcsearch')
  File "/usr/local/lib/python3.6/dist-packages/selenium/webdriver/remote/webdriv                                                                                      er.py", line 564, in find_element_by_class_name
    return self.find_element(by=By.CLASS_NAME, value=name)
  File "/usr/local/lib/python3.6/dist-packages/selenium/webdriver/remote/webdriv                                                                                      er.py", line 978, in find_element
    'value': value})['value']
  File "/usr/local/lib/python3.6/dist-packages/selenium/webdriver/remote/webdriv                                                                                      er.py", line 321, in execute
    self.error_handler.check_response(response)
  File "/usr/local/lib/python3.6/dist-packages/selenium/webdriver/remote/errorha                                                                                      ndler.py", line 242, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.NoSuchElementException: Message: no such element: Una                                                                                      ble to locate element: {"method":"class name","selector":"pcsearch"}
  (Session info: headless chrome=73.0.3683.86)
  (Driver info: chromedriver=2.46.628388 (4a34a70827ac54148e092aafb70504c4ea7ae9                                                                                      26),platform=Linux 4.4.0-135-generic x86_64)

2019-04-03 12:01:16,403 :: INFO :: root :: -------------------------PC----------                                                                                      ---------------
2019-04-03 12:01:18,103 :: INFO :: root :: Logging in account2@gmail.com...
2019-04-03 12:01:41,767 :: INFO :: root :: Search Start
2019-04-03 12:08:22,189 :: INFO :: root :: Stopped at search number 60
2019-04-03 12:08:35,613 :: INFO :: root :: Number of open offers: 8
```

The script then stops without going through any of account1 or account3's daily offers or searches

**Comment by darkconsole** [comment]
yeah it is currently stopping for me too after the first iteration of --pc. only difference i see is mine cant find the class edgesearch where yours cannot find pcsearch.

**Comment by xMNG** [comment]
I'm testing an update to the wait function for the flyout item. For the pc and edge search, are those daily points available to you in your region?

**Comment by darkconsole** [comment]
i experimented with this project for the first time today after another one i was using got deleted from github. pc mobile and quiz searches are all things that last one did and i can also do manually just by actually using bing properly.

tl;dr i don't understand the region question but i think so.

**Comment by ShoGinn** [comment]
The region question is that pcsearch is world and edge seems to be us. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by xMNG** [comment]
I pushed a fix for the pcpoints and removed the edge status, as it no longer appears during the points check status screen. Let me know if the errors continue, but the script is working well on my local machine.

**Comment by darkconsole** [comment]
aye latest pull is running well. watchin dem points go

**Comment by Bentechy66** [comment]
Will pull and report results

**Comment by Bentechy66** [comment]
The latest fix appears to be running well, all points have been received today an all three accounts.

**Comment by TravisDee67** [comment]
Latest fix is not working for me. 

I'm getting this error for PC search.
```
2019-04-04 11:45:12,932 :: ERROR :: root :: WebDriverException while executing pc portion
Traceback (most recent call last):
  File "ms_rewards.py", line 832, in <module>
    search(search_list)
  File "ms_rewards.py", line 481, in search
    if get_point_total(pc=True):
  File "ms_rewards.py", line 686, in get_point_total
    if not wait_until_visible(By.CLASS_NAME, 'pcsearch', 10):  # if object not found, return False
  File "ms_rewards.py", line 282, in wait_until_visible
    if browser.find_element(by=by_, value=selector):
  File "/usr/local/lib/python3.6/dist-packages/selenium/webdriver/remote/webdriver.py", line 966, in find_element
    'value': value})['value']
  File "/usr/local/lib/python3.6/dist-packages/selenium/webdriver/remote/webdriver.py", line 320, in execute
    self.error_handler.check_response(response)
  File "/usr/local/lib/python3.6/dist-packages/selenium/webdriver/remote/errorhandler.py", line 242, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {"method":"class name","selector":"pcsearch"}
```

and this error for mobile.
```
2019-04-04 12:06:49,421 :: INFO :: root :: WebDriverException while executing mobile portion
Traceback (most recent call last):
  File "ms_rewards.py", line 810, in <module>
    search(search_list, mobile_search=True)
  File "ms_rewards.py", line 475, in search
    if get_point_total(mobile=True):
  File "ms_rewards.py", line 686, in get_point_total
    if not wait_until_visible(By.CLASS_NAME, 'pcsearch', 10):  # if object not found, return False
  File "ms_rewards.py", line 282, in wait_until_visible
    if browser.find_element(by=by_, value=selector):
  File "/usr/local/lib/python3.6/dist-packages/selenium/webdriver/remote/webdriver.py", line 966, in find_element
    'value': value})['value']
  File "/usr/local/lib/python3.6/dist-packages/selenium/webdriver/remote/webdriver.py", line 320, in execute
    self.error_handler.check_response(response)
  File "/usr/local/lib/python3.6/dist-packages/selenium/webdriver/remote/errorhandler.py", line 242, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {"method":"class name","selector":"pcsearch"}
```

**Comment by Bentechy66** [comment]
I think this might be a level 1 issue not a script issue?

**Comment by xMNG** [comment]
@TravisDee67 can you confirm if the pcsearch class is present in the point url? Also what is your location.

**Comment by TravisDee67** [comment]
@xMNG I'm in the Digital Ocean US droplet. Not quite sure what you mean regarding the point URL.

I went to the "bepflyoutpage" and see pcsearch. The error is the following even for the mobile section.
```
selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {"method":"class name","selector":"pcsearch"}
```

I'll check again tomorrow to see if it works again.

@Bentechy66 I'm a level two.
